docs/guides/Building Token Factories/creatingmetadata.md

```diff
@@ -0,0 +1,204 @@
+
+To let the system identify those different elements you must provide a `manifest.json` where you describe your images by giving a textual descriptions and a path to them in the `uniq_factory` folder.
+
+Aside the requirement above, you can put anything else in your `uniq_factory` folder. Those files will be part of the NFT owned by the user and will be the asset(s) the user bought.
+
+It is recommended to use the following folder setup for ease of use.
+
+```
+example/ 📁 (This is your Root Folder)
+|- factory.csv
+|- tokens.csv
+|- manifest.json
+|
+|- factory/ 📁
+|   |- square.png
+|   |- uniq.png
+|   |- gallery/ 📁
+|       |- media-01.png
+|
+|- tokens/ 📁
+    |- 1 📁
+    |   |- uniq.png
+    |   |- gallery/ 📁
+    |       |- media-01.png
+    |
+    |- 2 📁
+    |   |- uniq.png
+    |   |- gallery/ 📁
+    |       |- media-01.png
+```
+
+However, the folder structure can be modified and replaced with a single folder that contains CSV files and relevant images/media files.
+
+### Setup CSV Template:
+
+1.  Use the following CSV template: [Google Sheets Metadata Tool Template](https://docs.google.com/spreadsheets/d/1Gi0iuJis-riKkyhYgMRYVnhGD6PbjvBv8U7lhRwdhNk/edit?usp=sharing) (Visit the link and go to `File -> Make a Copy` to make a copy in your own Google Drive).
+2.  Open your copied CSV template in your Google Drive and begin modifications of individual fields. (Read the first sheet for information regarding the template).
+3.  Export both `factory` and `tokens` sheets as CSV files. `File -> Download -> csv`
+4.  Rename factory file to `factory.json` and place in your root folder. (See [Folder Structure](#folder-structure) Above)
+5.  Rename token file to `tokens.json` and place in your root folder.
+
+### Image/Media Pathing:
+
+There are two ways to link to an image/media file in your CSV file. One is relative path, and the other is an external HTTP or HTTPS address.
+
+-   Relative Path: `./tokens/1/image.png` (Relative to your root folder)
+-   External: `https://some-external-website.com/tokens/1/image.png`
+
+For example, in the above structure, your CSV file should have referenced the images like this:
+
+![CSV example](/images/token-factories/csv_example.png)
+
+If the paths in your CSV are incorrect, then when you try to run the metadata tool against the files which are online, you will get errors and your images in your `uploads.json` will show `undefined`. This must be fixed before you move on to the next step.
+
+### Supported Media Types:
+
+The tool currently only supports the follow media types:
+
+-   jpg / jpeg
+-   png
+-   bmp
+-   gif
+-   webp
+-   mp4
+-   webm
+-   json
+
+### Running the tool:
+
+Once the CSV files and all related images/media files are present in the folder, you can process the folder for JSON creation.
+
+1. Drag & Drop the Folder onto the Executable
+
+    a. Watch the console window for errors.
+
+    b. If errors have occurred, you can check the report generated after closing the application
+
```

Comment:
Is the guide for windows, mac, or linux users, or all of them. Cause drapg and drop doesn't work on ubuntu 22 when I test it. I can only use command line to do it.

Reply:
That's good to know. I didn't test Linux.